docs(components): Add info about `extensions` and `pathPrefix` keys (#8930)
nuxt/framework: docs/content/2.guide/2.directory-structure/1.components.md

</template>
```

## Component extensions

> By default, any file with extension specified in [extensions key of `nuxt.config.ts`](/api/configuration/nuxt-config#extensions) treats as a component. \

Reviewer: ```suggestion
By default, any file with an extension specified in the [extensions key of `nuxt.config.ts`](/api/configuration/nuxt-config#extensions) is treated as a component.
```

> If you need to restrict file extensions that should be registered as components, use the extended form of components directory declaration and its `extensions` key:

Reviewer: ```suggestion
If you need to restrict the file extensions that should be registered as components, you can use the extended form of the components directory declaration and its `extensions` key:
```

For clarity, we recommend that the component's filename matches its name. (So, in the example above, you could rename `Button.vue` to be `BaseFooButton.vue`.)
::

> If you want to auto-import components based only on its name, not path, then you need to set `pathPrefix` option to `false` using extended form of configuration object:

Reviewer: ```suggestion
If you want to auto-import components based only on its name, not path, then you need to set `pathPrefix` option to `false` using extended form of the configuration object:
```

```diff
export default defineNuxtConfig({
  components: [
    {
      path: '~/components/',
+     pathPrefix: false,
    },
  ],
});
```

> This registers the components using the same strategy as used in Nuxt 2.

Reviewer 2: I think you should explain this more clearly cause not every Nuxt 3 devs have used Nuxt 2.

Author: I think this is explained well in a paragraph before code block. 
The reason I mention Nuxt 2 is [this comment](https://github.com/nuxt/framework/issues/1044#issuecomment-943562072) (and some next ones).

If developer used Nuxt2 then he will understand, and if he didn't then he has nothing to compare with anyway.
Perhaps this mention is simply redudant, especially if there is a focus on new members of the community.

Reviewer 2: > I think this is explained well in a paragraph before code block. The reason I mention Nuxt 2 is [this comment](https://github.com/nuxt/framework/issues/1044#issuecomment-943562072) (and some next ones).
> 
> If developer used Nuxt2 then he will understand, and if he didn't then he has nothing to compare with anyway. Perhaps this mention is simply redudant, especially if there is a focus on new members of the community.

I understand your reason, but imo we should explain it since maybe it's the feature that someone have been looking for. Maybe at least an example compared between `pathPrefix: false` and `pathPrefix: true`?

> This registers the components using the same strategy as used in Nuxt 2.

Reviewer: ```suggestion
This registers the components using the same strategy as used in Nuxt 2. For example, `~/components/Some/MyComponent.vue` will be usable as `<MyComponent>` and not `<SomeMyComponent>`.
```
